### PR TITLE
Use formless default browsable API renderer

### LINKED ISSUE #970: Reporting HTML view `SELECT`s all media records
The reporting views for each media type should only accept `POST` requests. Currently, they allow `GET` requests and Django renders an HTML form for submitting a report. This form contains a dropdown of all identifiers for each media model (images and audio, currently), which is an _extremely_ expensive query that taxes our production database and threatens to take down Openverse.

These views are currently blocked in Cloudflare and set to redirect to the API homepage until we fix the problem. `POST` requests to the endpoints continue to work as expected.

## Proposed solutions

1. Try only allow POST methods in Django by adding `@action(detail=True, methods=["POST"])` above the `def report` view logic. I tried this and the page successfully errors and returns json, but still renders the HTML form!

  ```json
  {
      "detail": "Method \"GET\" not allowed."
  }
  ```
  
2. Find a way to modify and remove the form from the HTML view
3. Something I haven't thought of yet.

### PR DESCRIPTION
**Pull request by sarayourfriend**
## Fixes

Fixes #970 by @zackkrida 

## Description

The view in question in the issue is the browsable API view. You'll get a 405 method not allowed, but the form below is the culprit here.

The new renderer added in this PR removes this form from all the views to prevent this kind of thing happening in any of our browsable views. This has the benefit of still maintaining the browsable API views which are quite useful to all developing the API or using it for their own purposes.

## Testing Instructions

Run the branch (`just up`) and visit the browsable API views. Test something like http://localhost:50280/v1/images/2982c0bf-9c26-4543-a712-4d50a117ded8/report/ specifically. On `main` you'll see a form underneath the JSON blob reporting the method not allowed error. On this branch, you'll see no form.

## Checklist

- [x] My pull request has a descriptive title (not a vague title like
      `Update index.md`).
- [x] My pull request targets the _default_ branch of the repository (`main`) or
      a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added or updated tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible
      errors.

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: api/catalog/settings.py [hunk 1]
```diff
@@ -145,7 +145,7 @@
     "DEFAULT_VERSIONING_CLASS": "rest_framework.versioning.URLPathVersioning",
     "DEFAULT_RENDERER_CLASSES": (
         "rest_framework.renderers.JSONRenderer",
-        "rest_framework.renderers.BrowsableAPIRenderer",
```
Comment (dhruvkb) [inline api/catalog/settings.py:148]: I'm curious to know why not turn off the browsable API in production.

**Review by dhruvkb — APPROVED** [review]
LGTM!

File: api/catalog/settings.py (hunk 1 above)
Comment (zackkrida) [reply api/catalog/settings.py:148]: We discussed in Slack, but generally we find it quite useful while browsing the API in a browser!

Comment (dhruvkb) [reply api/catalog/settings.py:148]: In production?

**Review by zackkrida — APPROVED** [review]
LGTM. Confirmed that this does not execute the `SELECT * from images` query as well.

**Pull request merged**